### PR TITLE
kconfig: cleanup

### PR DESCRIPTION
**Pull request by lemrey**
The Kconfig tree is growing and there aren't many guidelines. As a consequence, it looks a bit messy.
This is an attempt to clean it up a bit, have started with some uncontroversial changes (I hope) ~but I plan to add more commits~

### CODE REVIEW

**Comment by lemrey** [comment]
@rlubos , @tejlmand what do you think?
@joerchan can we move the TFM entries for the UART pin config?

**Review by joerchan — APPROVED** [review]
Yeah, no problem moving the TF-M Kconfig entries for the UART pins.

**Comment by joerchan** [comment]
@lemrey You got some nitpicking from the compliance check.

File: subsys/debug/Kconfig
Comment (jayteemo) [inline subsys/debug/Kconfig:12]: is a newline needed?

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]
LGTM

File: subsys/debug/Kconfig
Comment (tejlmand) [reply subsys/debug/Kconfig:12]: yes, there should be a newline here.
@lemrey

File: samples/Kconfig
Comment (tejlmand) [inline samples/Kconfig:109]: this is set by CMake on the child image here:
https://github.com/nrfconnect/sdk-nrf/blob/51167148f0f2158096e9fb54a02337f6f3737a22/cmake/multi_image.cmake#L225

For this to work, this setting needs a prompt, else CMake cannot set it.
CMake is the "user" in this case.

File: Kconfig.nrf
Comment (tejlmand) [inline Kconfig.nrf:26]: this is extending the upstream choice menu from here:
https://github.com/nrfconnect/sdk-zephyr/blob/df384c23a1a463623a6b0bb57ce1a726515a13ec/modules/mbedtls/Kconfig#L23

to allow users to use nRF Security.

To my knowledge this cannot be removed, but @frkv can confirm.

**Review by tejlmand — CHANGES_REQUESTED** [review]
some cleanup cannot be done.

Also, why is `CI-disable` label added ?
There are changes here that i'm confident will break CI because certain settings are no longer available for configurations.
So cannot approve this PR without a CI run.

**Review by simensrostad — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

File: samples/Kconfig
Comment (lemrey) [reply samples/Kconfig]: Thank you, I opted for putting them into their own menu so they no longer mix with other top-level menu entries.

File: Kconfig.nrf
Comment (lemrey) [reply Kconfig.nrf:26]: This was moved to the Kconfig file in `/subsys`

File: subsys/debug/Kconfig
Comment (lemrey) [reply subsys/debug/Kconfig]: Thank you, updated

**Comment by lemrey** [comment]
@tejlmand I added the `CI-disable` tag because this spins every test plan possible. I did not want to do that unless the changes looked good to most people to avoid using too much CPU time on the servers. Of course we can let the tests run once before we merge.

**Review by SeppoTakalo — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]

**Review by frkv — APPROVED** [review]
LGTM

**Review by sebastiandraus — APPROVED** [review]

File: Kconfig.nrf
Comment (tejlmand) [reply Kconfig.nrf:26]: didn't notice that, thanks for the update.

**Comment by tejlmand** [comment]
> Of course we can let the tests run once before we merge.

given the nature of the changes, that would be very nice.
and I see you already removed the label.

File: Kconfig.nrf
Comment (frkv) [reply Kconfig.nrf:26]: Yes. I saw this PR and approved it because the configs aren't removed, they are just moved

When I think about it there is one thing that may be important to verify and that is that the subsys n use is not taken in before Zephyr (which we are overriding)

**Comment by rlubos** [comment]
> > Of course we can let the tests run once before we merge.
> 
> given the nature of the changes, that would be very nice. and I see you already removed the label.

The tests were already executed for this PR (https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf/job/master/26606/), some jobs needed a rerun but eventually, everything passed.

File: Kconfig.nrf
Comment (tejlmand) [reply Kconfig.nrf:26]: the commit message says:
> Try to clean up the top-level Kconfig menu.

but the reality is that this setting is still located in the same top-level nrf menu, just a bit further down now.
`sdk-nrf/main`:
image

PR #7996:
image

so the top-level Kconfig menu is not really cleaned up.

Was this your intention with the PR ?

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Comment by lemrey** [comment]
@tejlmand no, it wasn't. I originally had one more commit locally, where I further grouped entries creating one menu for "Subsystems" and put mbedTLS there. I did not push that commit to this PR, so I am the mbedTLS commit as well because it does not make much sense.

**Comment by ns-tolu** [comment]
@tejlmand could you please review recent changes in this PR?

**Review by tejlmand — APPROVED** [review]
lgtm, thanks for the cleanup.

**Pull request merged**
